client-server-server/src/main/scala/zio/raft/server/RaftServer.scala

```diff
@@ -0,0 +1,473 @@
+      
+      override def handle(
+        event: StreamEvent,
+        transport: ZmqServerTransport,
+        config: ServerConfig,
+        raftActionsOut: Queue[RaftAction]
+      ): UIO[ServerState] = {
+        event match {
+          case StreamEvent.IncomingClientMessage(routingId, message) =>
+            handleClientMessage(routingId, message, transport, raftActionsOut)
+          
+          case StreamEvent.Action(ServerAction.StepUp(_)) =>
+            ZIO.succeed(this)
+          
+          case StreamEvent.Action(ServerAction.StepDown(leaderId)) =>
+            for {
+              _ <- ZIO.logInfo("Lost leadership, closing all sessions")
+              _ <- sessions.closeAll(transport, leaderId)
+            } yield Follower()
+          
+          case StreamEvent.Action(ServerAction.SendResponse(sessionId, response)) =>
+            sessions.getRoutingId(sessionId) match {
+              case Some(routingId) =>
+                transport.sendMessage(routingId, response).orDie.as(this)
+              case None =>
+                ZIO.logWarning(s"Cannot send response - session $sessionId not found").as(this)
+            }
+          
+          case StreamEvent.Action(ServerAction.SendServerRequest(sessionId, request)) =>
+            sessions.getRoutingId(sessionId) match {
+              case Some(routingId) =>
+                transport.sendMessage(routingId, request).orDie.as(this)
+              case None =>
+                ZIO.logWarning(s"Cannot send server request - session $sessionId not found").as(this)
+            }
+          
+          case StreamEvent.CleanupTick =>
+            for {
+              now <- Clock.instant
+              (expiredIds, newSessions) = sessions.removeExpired(now)
+              _ <- ZIO.foreachDiscard(expiredIds) { sessionId =>
+                for {
+                  _ <- raftActionsOut.offer(RaftAction.ExpireSession(sessionId))
+                  routingIdOpt = sessions.getRoutingId(sessionId)
+                  _ <- routingIdOpt match {
+                    case Some(routingId) =>
+                      transport.sendMessage(routingId, SessionClosed(SessionCloseReason.SessionTimeout, None)).orDie
+                    case None => ZIO.unit
+                  }
+                } yield ()
+              }
+            } yield copy(sessions = newSessions)
+        }
+      }
+      
+      private def handleClientMessage(
+        routingId: RoutingId,
+        message: ClientMessage,
+        transport: ZmqServerTransport,
+        raftActionsOut: Queue[RaftAction]
+      ): UIO[ServerState] = {
+        message match {
+          case CreateSession(capabilities, nonce) =>
+            for {
+              sessionId <- SessionId.generate()
+              now <- Clock.instant
+              _ <- raftActionsOut.offer(RaftAction.CreateSession(sessionId, capabilities))
+              _ <- transport.sendMessage(routingId, SessionCreated(sessionId, nonce)).orDie
```

Comment:
we should not send the response yet, because the node didn't commit this yet to the state machine. We should instead wait for a action, notifying us that the session was created, only then we should send the response.

We should also somehow, record that the session is not yet final, but still keep the routing id to session id. We should not accept messages on that session for example.